src/scalar/fixed_size_binary.rs

```diff
@@ -0,0 +1,52 @@
+use crate::datatypes::DataType;
+
+use super::Scalar;
+
+#[derive(Debug, Clone, PartialEq)]
+/// The [`Scalar`] implementation of fixed size binary ([`Option<Box<[u8]>>`]).
+pub struct FixedSizeBinaryScalar {
+    value: Option<Box<[u8]>>,
+    data_type: DataType,
+}
+
+impl FixedSizeBinaryScalar {
+    /// Returns a new [`FixedSizeBinaryScalar`].
+    /// # Panics
+    /// iff
+    /// * the `data_type` is not `FixedSizeBinary`
+    /// * the size of child binary is not equal
+    #[inline]
+    pub fn new<P: Into<Vec<u8>>>(data_type: DataType, value: Option<P>) -> Self {
+        Self {
```

Comment:
```suggestion
        assert_eq!(data_type.to_physical_type(), crate::datatypes::PhysicalType::FixedSizeBinary);
        Self {
```

so that if the value us null, we still have a valid physical type on it.